stix_shifter_modules/ibm_security_verify/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,323 @@
+import logging
+import re
+from typing import Union
+
+from stix_shifter_utils.stix_translation.src.json_to_stix import observable
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import (
+    CombinedComparisonExpression, CombinedObservationExpression,
+    ComparisonComparators, ComparisonExpression, ComparisonExpressionOperators,
+    ObservationExpression, ObservationOperators, Pattern)
+from stix_shifter_utils.stix_translation.src.utils.transformers import \
+    TimestampToMilliseconds
+
+# Source and destination reference mapping for ip and mac addresses.
+# Change the keys to match the data source fields. The value array indicates the possible data type that can come into from field.
+REFERENCE_DATA_TYPES = {
+    "ipaddr": ["ipv4", "ipv4_cidr", "ipv6", "ipv6_cidr"],
+    "proxy_ip": ["ipv4", "ipv4_cidr"],
+}
+REFERENCE_FILTER_TYPE = ["user-account", "ipv4-addr", "domain-name", "x-oca-event"]
+REFERENCE_EXCLUDE_FILTER_TYPE = ["category"]
+logger = logging.getLogger(__name__)
+
+
+class QueryStringPatternTranslator:
+    QUERIES = []
+    comparator_lookup = {
+        "ComparisonExpressionOperators.And": "&",
```

Comment:
These keys shouldn't be strings, they are Classes that you are importing at the top.